### PR TITLE
feat(providers): Add possibility to Onesignal choose version api

### PR DESCRIPTION
**Pull request by suplere**
### What changed? Why was the change needed?

This PR add possibility to Onesignal provider choose between:
a) player model
b) user model

[User model documentation](https://documentation.onesignal.com/docs/users) 

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

**Comment by L-U-C-K-Y** [comment]
@suplere nice, you created a new one, I actually also just did, what do you think about my MR?
I like that you use the new API, I just changed the key with an override, a bit simpler: https://github.com/novuhq/novu/pull/6979

**Comment by L-U-C-K-Y** [comment]
@suplere I would vote for your version! I'll close mine after this is merged.
Thank you so much

@scopsy sorry for tagging you again, but is the better MR, would highly appreciate if we could get it merged to unblock our users

**Review by Michael-JobDone — COMMENTED** [review]
👍

**Comment by suplere** [comment]
This solution use new API. This allows you to use the new and recommended Onesignal API. The old API (with player model) may be marked as obsolete or unsupported over time.

**Comment by L-U-C-K-Y** [comment]
> This solution use new API. This allows you to use the new and recommended Onesignal API. The old API (with player model) may be marked as obsolete or unsupported over time.

Absolutely, I closed mine

File: packages/shared/src/consts/providers/credentials/provider-credentials.ts
Comment (L-U-C-K-Y) [inline packages/shared/src/consts/providers/credentials/provider-credentials.ts]: @suplere do we need to use `CredentialsKeyEnum.Domain` here as we use `this.apiVersion = config.domain;` in the provider?

Comment (scopsy) [reply packages/shared/src/consts/providers/credentials/provider-credentials.ts]: Yes, I have updated this to be apiVersion and passed the proper info from the handler

File: packages/providers/src/lib/push/one-signal/one-signal.provider.ts
Comment (scopsy) [inline packages/providers/src/lib/push/one-signal/one-signal.provider.ts:59]: @L-U-C-K-Y, do you know by chance if the API body signature is similar to both modes and API endpoints?

Comment (scopsy) [reply packages/providers/src/lib/push/one-signal/one-signal.provider.ts:59]: I'm just wondering if we even need to change the BASE URL at all

Comment (L-U-C-K-Y) [reply packages/providers/src/lib/push/one-signal/one-signal.provider.ts:59]: @scopsy yes that would still be required if we want to keep "include_player_ids" in the older model.
Your change looks all good for me 😄 

**v9**
See here the API Documentation for v9 (old):
https://documentation.onesignal.com/v9.0/reference/create-notification

image

V9 does **not** have the key `include_aliases` (was named: `include_external_user_ids` and still had the concept of player ids and therefore had `include_player_ids`.

**v11.6**
And here 11.6 (new):
https://documentation.onesignal.com/reference/push-notification

image

V11.6 has `include_aliases` where we can send the external ids, but **no** `include_player_ids`.
In the new versions, player ids are not used anymore and the recommended way is the external id:

https://documentation.onesignal.com/docs/users

_It is always recommended to use the External ID to identify users because this is the only way to track a user across all their subscriptions._

**Review by suplere — COMMENTED** [review]
Hello, it is looks better than my version.

File: packages/providers/src/lib/push/one-signal/one-signal.provider.ts
Comment (L-U-C-K-Y) [inline packages/providers/src/lib/push/one-signal/one-signal.provider.ts]:
```suggestion
        ? {
            include_aliases: {
              external_id: options.target,
            },
            target_channel: 'push',
          }
```

I just noticed in the docs that we would need to include the target_channel for the new API endpoint:

image

File: packages/providers/src/lib/push/one-signal/one-signal.providerV2.spec.ts
Comment (L-U-C-K-Y) [inline packages/providers/src/lib/push/one-signal/one-signal.providerV2.spec.ts:58]:
```suggestion
      include_aliases: {
        external_id: ['userId'],
      },
      target_channel: 'push',
```

I just noticed in the docs that we would need to include the target_channel for the new API endpoint:

image

**Comment by suplere** [comment]
Is there some problem ?

**Comment by scopsy** [comment]
@suplere could you take a look on @L-U-C-K-Y suggestions?

File: packages/providers/src/lib/push/one-signal/one-signal.provider.ts
Comment (suplere) [reply packages/providers/src/lib/push/one-signal/one-signal.provider.ts]: Yes this si good addition target channell pu#h

**Review by suplere — COMMENTED** [review]
This si good addition target channel push

**Comment by L-U-C-K-Y** [comment]
> This si good addition target channel push

@suplere can you please make the changes? It's your MR. You can also just merge this one: https://github.com/suplere/novu/pull/1

Then we're all good!

**Comment by L-U-C-K-Y** [comment]
@scopsy changes are merged, LGTM! 😄

**Review by scopsy — APPROVED** [review]

**Comment by KrunchMuffin** [comment]
Whats the status on this? Can we get the conflicts resolved?

**Comment by L-U-C-K-Y** [comment]
> Whats the status on this? Can we get the conflicts resolved?

It's fine from my side, maybe someone of the maintainers @scopsy or someone else could help here 😄

**Comment by scopsy** [comment]
Due to permission issues on the fork, I've resolved the errors on https://github.com/novuhq/novu/pull/7270 and merged it 🎉

Thank you so much for the collaboration and my apologies for the delayed response on this 🙏

**Pull request closed without merging**
